### PR TITLE
test: Navigation Bar

### PR DESCRIPTION
**Pull request by DE7924**

### CODE REVIEW

File: lib/src/components/navigation bar/navigation_bar.dart [hunk 1]
```diff
@@ -251,9 +264,11 @@ class _NavigationItem extends ZetaStatelessWidget {
       child: InkResponse(
         borderRadius: context.rounded ? Zeta.of(context).radius.rounded : Zeta.of(context).radius.none,
         onTap: onTap,
+        hoverColor: colors.surfaceHover,
+        highlightShape: BoxShape.rectangle,
         child: Semantics(
           button: true,
-          explicitChildNodes: true,
+          excludeSemantics: true,
```
Comment (thelukewalton) [inline lib/src/components/navigation bar/navigation_bar.dart:271]: Can you revert this change please. We need ```explicitChildNodes: true,``` so that if the icon has a notification badge the semantics can see it

File: lib/src/components/navigation bar/navigation_bar.dart [hunk 2]
```diff
@@ -278,11 +293,9 @@ class _NavigationItem extends ZetaStatelessWidget {
                 ),
                 SizedBox(height: Zeta.of(context).spacing.small),
                 if (item.label != null)
-                  ExcludeSemantics(
-                    child: Text(
-                      item.label!,
-                      style: Theme.of(context).textTheme.labelSmall?.copyWith(color: elementColor),
-                    ),
+                  Text(
```
Comment (thelukewalton) [inline lib/src/components/navigation bar/navigation_bar.dart:296]: This is linked to the above - we want a badge or any other widget to be individual, but the text is part of the parent so we need to exclude it so that it is not duplicated

File: test/src/components/navigation_bar/navigation_bar_test.dart [hunk 3]
```diff
@@ -0,0 +1,482 @@
+import 'package:flutter/gestures.dart';
```
Comment (thelukewalton) [inline test/src/components/navigation_bar/navigation_bar_test.dart:1]: Is there any test that specifically tests [UX-1096](https://jira.zebra.com/browse/UX-1096) - something like tapping off center that would have failed before those changes but would now pass?

File: lib/src/components/navigation bar/navigation_bar.dart (hunk 2 above)
Comment (DE7924) [reply lib/src/components/navigation bar/navigation_bar.dart:296]: So wrap this text in excludeSemantics

Comment (thelukewalton) [reply lib/src/components/navigation bar/navigation_bar.dart:296]: yes please

File: lib/src/components/navigation bar/navigation_bar.dart (hunk 1 above)
Comment (DE7924) [reply lib/src/components/navigation bar/navigation_bar.dart:271]: When the explicitChildNodes is set to true the notification dot the notification indicator fails accessibility. In fact the default color medium notification indicator always fails accessibility due to text contrast ratios. The large however does not fail: 
image

Comment (DE7924) [reply lib/src/components/navigation bar/navigation_bar.dart:271]: The indicator fails contrast ratios only with small sizes: 
image
I'll raise the issue to the designers

Comment (thelukewalton) [reply lib/src/components/navigation bar/navigation_bar.dart:271]: I think there must be some other issue with the test here - the  colors being tested are  #E0E3E9 (light grey) and #FFFFFF (white) - for the notification indicator the background should be red?

Comment (DE7924) [reply lib/src/components/navigation bar/navigation_bar.dart:271]: Perhaps some overlap with the border?
